src/scroll-timeline-base.js

```diff
@@ -510,3 +510,161 @@ export class ScrollTimeline {
     return true;
   }
 }
+
+function getScrollParent(node) {
+  if (node === undefined)
+    return undefined;
+
+  if (node.scrollHeight > node.clientHeight ||
+      node.scrollWidth > node.clientWidth) {
+    return node;
+  } else {
+    return getScrollParent(node.parentNode);
+  }
+}
+
+// https://fantasai.inkedblade.net/style/specs/scroll-animations-1/#view-progress-timelines
+export class ViewTimeline extends ScrollTimeline {
+  // As specced, ViewTimeline has a subject and a source, but
+  // ViewTimelineOptions only has source. Furthermore, there is a strict
+  // relationship between subject and source (source is nearest scrollable
+  // ancestor of subject).
+
+  // Proceeding under the assumption that subject will be added to
+  // ViewTimelineOptions. Inferring the source from the subject if not
+  // explicitly set.
+  constructor(options) {
+    if (options.subject && !options.source) {
+      options.source = getScrollParent(options.subject.parentNode);
```

Comment:
We rely on the source being set in order to attach the scroll event listener.  I can add a TODO to add a mutation observer to watch for style changes that would affect the resolution of 'source'.

Reply:
Adding a TODO is a good idea. We won't be able to observe style changes but we could at least re-evaluate the scroll source on a regular cadence. I have some ideas how to do this in a way that isn't too expensive - it will get slightly out of date but fix itself quickly.